<launch>

    <arg name="start_state" default="Start" />
    <arg name="map_path" default="$(find sigverse_hsrb_nav)/map/map.yaml" />
    <arg name="use_map_topic" default="false" />
    <arg name="use_map_server" default="true" />
    <arg name="use_teleop" default="false" />
    <arg name="debug" default="false" />
    <arg name="use_rviz" default="true" />
    <arg name="world_model_name" default="interactive_cleanup"/>
    <arg name="upload_robot_model_launch" default="$(find hsrb_description)/robots/upload_hsrb.launch" />
    <arg name="wait_to_ready" default="true" />

    <arg name="is_sigverse" default="true"/>
    <arg name="human_max_distance" default="10.0"/>
    <arg name="base_prompt" default="handyman"/>

    <param name="/handyman/wait_to_ready" value="$(arg wait_to_ready)" />

    <node name="state_machine" pkg="tamhome_handyman" type="task_node.py" output="screen">
        <param name="start_state" value="$(arg start_state)" />
    </node>

    <!-- スキル理解サーバ -->
    <node name="skill_parser" pkg="tamhome_task_parser" type="task_parser_service.py" output="screen" respawn="true">
        <param name="base_prompt" value="$(arg base_prompt)" />
    </node>

    <!-- 人物認識 -->
    <!-- <include file="$(find tam_mmaction2)/launch/simple_test.launch">
        <arg name="is_sigverse" value="$(arg is_sigverse)"/>
        <arg name="max_distance" value="$(arg human_max_distance)"/>
        <arg name="use_rviz" value="false" />
    </include> -->

    <!-- 人に注目 -->
    <!-- <node pkg="tamhome_pointing_estimation" type="focus_person.py" name="focus_person_node" output="screen" respawn="true"/> -->

    <!-- 指差し推定 -->
    <!-- <node pkg="tamhome_pointing_estimation" type="pointing_estimation.py" name="pointing_estimation" output="screen" respawn="true"/> -->

    <!-- world model 読み込み -->
    <!-- <include file="$(find tam_dynamic_map)/launch/load_model.launch">
        <arg name="world_model_name" value="$(arg world_model_name)"/>
    </include> -->

    <group if="$(arg use_rviz)">
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find tamhome_handyman)/config/handyman.rviz" />
    </group>

</launch>